src/algebra/group/units.lean

```diff
@@ -403,6 +413,28 @@ begin
   simp [h.unit_spec]
 end
 
+section monoid
+variables [monoid M] {a b c : M}
+
+@[to_additive] lemma is_unit.mul_left_inj (h : is_unit a) : b * a = c * a ↔ b = c :=
+h.unit.mul_left_inj
+
+@[to_additive] lemma is_unit.mul_right_inj (h : is_unit a) : a * b = a * c ↔ b = c :=
+h.unit.mul_right_inj
```

Comment:
Note that this invokes `choice` needlessly vs using `obtain` or `let`.

Reply:
Do we care?

Reply:
I think the answer is "mostly no", but I think a lot of the `is_unit` proofs do avoid it.